examples/library/src/Server.hs

```diff
@@ -0,0 +1,107 @@
+import           Control.Monad.IO.Class            (MonadIO (..))
+import           Control.Monad.Logger
+import           Data.Conduit
+import           Data.Maybe                        (fromJust)
+import qualified Data.Text                         as T
+import           Database.Persist
+import           Database.Persist.Sqlite
+import           Mu.Adapter.Persistent             (runDb)
+import           Mu.GraphQL.Quasi
+import           Mu.GraphQL.Server
+import           Mu.Schema
+import           Mu.Server
+import           Network.Wai.Handler.Warp          (run)
+import           Network.Wai.Middleware.AddHeaders (addHeaders)
+
+import           Schema
+
+main :: IO ()
+main = do
+  putStrLn "starting GraphQL server on port 8000"
+  let hm = addHeaders [
+             ("Access-Control-Allow-Origin", "*")
+           , ("Access-Control-Allow-Headers", "Content-Type")
+           ]
+  runStderrLoggingT $
+    withSqliteConn @(LoggingT IO) ":memory:" $ \conn -> do
+      runDb conn $ runMigration migrateAll
+      liftIO $ run 8000 $ hm $
+        graphQLApp (libraryServer conn)
+                   (Proxy @('Just "Query"))
+                   (Proxy @('Just "Mutation"))
+                   (Proxy @('Just "Subscription"))
+
+type ObjectMapping = '[
+    "Book"   ':-> Entity Book
+  , "Author" ':-> Entity Author
+  ]
+
+libraryServer :: forall i.
+                 SqlBackend
+              -> ServerT ObjectMapping i Library ServerErrorIO _
+libraryServer conn
+  = resolver ( object @"Book"     ( field  @"id"       bookId
+                                  , field  @"title"    bookTitle
+                                  , field  @"author"   bookAuthor )
+             , object @"Author"   ( field  @"id"       authorId
+                                  , field  @"name"     authorName
+                                  , field  @"books"    authorBooks )
+             , object @"Query"    ( method @"authors"  allAuthors
+                                  , method @"books"    allBooks )
+             , object @"Mutation" (method @"newAuthor" newAuthor)
+             , object @"Subscription" ( method @"allBooks" allBooksConduit )
+             )
+  where
+    bookId :: Entity Book -> ServerErrorIO Integer
+    bookId (Entity (BookKey k) _) = pure $ toInteger k
+    bookTitle :: Entity Book -> ServerErrorIO T.Text
+    bookTitle (Entity _ Book { bookTitle = t }) = pure t
+    bookAuthor :: Entity Book -> ServerErrorIO (Entity Author)
+    bookAuthor (Entity _ Book { bookAuthor = a })
+      = runDb conn $ Entity a . fromJust <$> get a
+    authorId :: Entity Author -> ServerErrorIO Integer
+    authorId (Entity (AuthorKey k) _) = pure $ toInteger k
+    authorName :: Entity Author -> ServerErrorIO T.Text
+    authorName (Entity _ Author { authorName = t }) = pure t
+    authorBooks :: Entity Author -> ServerErrorIO [Entity Book]
+    authorBooks (Entity a _)
+      = runDb conn $ selectList [BookAuthor ==. a] [Asc BookTitle]
```

Comment:
Wow this is so nice and convenient! (I guess these are operators from `persistent`, right?) 🎉

Reply:
Yep, that's how you query stuff using `persistent`. Really magical 🧙